Applications/web_portal/web_app/library/source/pages/account_page/risk_page/risk_parameters_view.tsx

```diff
@@ -10,11 +14,297 @@ interface Properties {
+                      MINUTE
+                    </span>
+                  </VBoxLayout>
+                  <Padding size='10px'/>
+                  <Center height='34px' className={
+                      css(RiskParametersView.TRANSITION_TIME_STYLE.colon)}>
+                    :
+                  </Center>
+                  <Padding size='10px'/>
+                  <VBoxLayout>
+                    <IntegerInputBox min={0} max={59} value={seconds}
+                      padding={2}
+                      className={css(RiskParametersView.STYLE.inputBox)}
+                      onChange={(value) => this.onTransitionTimeChange(
+                        value, TimeUnit.SECONDS)}/>
+                    <Padding size='10px'/>
+                    <span className={
+                        css(RiskParametersView.
+                          TRANSITION_TIME_STYLE.label)}>
+                      SECOND
+                    </span>
+                  </VBoxLayout>
+                </HBoxLayout>
+              </VBoxLayout>
+              <Padding/>
+            </HBoxLayout>
+          </VBoxLayout>
+        </VBoxLayout>
+        <Padding/>
+      </HBoxLayout>);
+  }
+
+  private static getBreakpoint(): Breakpoint {
+    const screenWidth = window.innerWidth ||
+      document.documentElement.clientWidth ||
+      document.getElementsByTagName('body')[0].clientWidth;
+    if(screenWidth <= 767) {
+      return Breakpoint.SMALL;
+    } else if(screenWidth > 767 && screenWidth <= 1035) {
+      return Breakpoint.MEDIUM;
+    } else {
+      return Breakpoint.LARGE;
+    }
+  }
+
+  private onScreenResize(): void {
+    const newBreakpoint = RiskParametersView.getBreakpoint();
+    if(newBreakpoint !== this.state.breakpoint) {
+      this.setState({breakpoint: newBreakpoint});
+    }
   }
+
+  private onCurrencyChange(value: Nexus.Currency) {
+    this.props.parameters.currency = value;
+    this.forceUpdate();
+  }
+
+  private onBuyingPowerChange(value: Nexus.Money) {
+    this.props.parameters.buyingPower = value;
+  }
+
+  private onNetLossChange(value: Nexus.Money) {
+    this.props.parameters.netLoss = value;
+  }
+
+  private onTransitionTimeChange(value: number, timeUnit: TimeUnit) {
+    const timeJSON = this.props.parameters.transitionTime.split();
+    const newTimeJSON = (() => {
+      switch (timeUnit) {
+        case TimeUnit.HOURS:
+          return {
+            hours: value,
+            minutes: timeJSON.minutes,
+            seconds: timeJSON.seconds
+          };
+        case TimeUnit.MINUTES:
+          return {
+            hours: timeJSON.hours,
+            minutes: value,
+            seconds: timeJSON.seconds
+          };
+        case TimeUnit.SECONDS:
+          return {
+            hours: timeJSON.hours,
+            minutes: timeJSON.minutes,
+            seconds: value
+          };
+        }
+    })();
+    this.props.parameters.transitionTime = Beam.Duration.fromJson(newTimeJSON);
```

Comment:
fromJSON unfortunately doesn't work like this, although this is a good idea.

You will need to do something like Beam.Duration.HOUR.multiply(newTime.hours) + Beam.Duration.MINUTE.multiply(newTime.minutes) + ...